src/api/torii-endpoints.md

```diff
@@ -0,0 +1,370 @@
+communication after the WebSocket handshake. Then the server sends a [`BlockMessage`](/api/data-model-schema#blockmessage).
+Messages are encoded with the [Parity Scale Codec](#parity-scale-codec).
+
+#### Notes
+
+Via this endpoint, the client first provides the starting block number (i.e. height) in the subscription request. After
+sending the confirmation message, the server starts streaming all the blocks from the given block number up to the current
+block and continues to stream blocks as they are added to the blockchain.
+
+### Get Configuration
+
+::: warning
+
+TODO: Will change as part of the config RFC
+
+:::
+
+- **Protocol:** HTTP
+- **Method:** `GET`
+- **Endpoint:** `/configuration`
+
+There are 2 possible request bodies and according responses.
+
+#### Get Configuration Value as JSON
+
+- **Request Body:**
+  ```json
+  "Value"
+  ```
+- **Responses:**
+  - `200 OK`: Configuration value as JSON
+
+#### Get Documentation of a Field
+
+- **Request Body:**
+  ```json
+  {
+    "Docs": ["a", "b", "c"]
+  }
+  ```
+  where "a.b.c" is a path of the field (e.g. `sumeragi.block_time_ms`).
+- **Responses:**
+  - `200 OK`: Field was found and either doc or value is returned in json body.
+  - `404 Not Found`: Field wasn't found
+
+::: tip
+
+If the requested field has more fields inside of it, then all the documentation for its inner members is returned as
+well. Here is an example for getting a field `a.b.c`:
+
+**Example:** getting top-level documentation for `Torii` and all the fields within:
+
+```bash
+curl -X GET -H 'content-type: application/json' http://127.0.0.1:8080/configuration \
+    -d '{"Docs" : ["torii"]} ' -i
+```
+
+:::
+
+### Configuration
+
+::: warning
+
+TODO: Will change as part of the config RFC
+
+:::
+
+- **Protocol:** HTTP
+- **Method:** `POST`
+- **Endpoint:** `/configuration`
+- **Expects:**
+  - **Body:** One configuration option is currently supported: `LogLevel`. It is set to the log-level in uppercase.
+    ```json
+    {
+      "LogLevel": "WARN"
+    }
+    ```
+    Acceptable values are `TRACE`, `DEBUG`, `INFO`, `WARN`, `ERROR`.
+- **Responses:**
+  - `200 OK`: Log level has changed successfully. The confirmed new log level is returned in the body.
+  - `400 Bad Request`: request body malformed.
+  - `500 Internal Server Error`: Request body valid, but changing the log level failed (lock contention).
+
+### Health
+
+- **Protocol:** HTTP
+- **Method:** `GET`
+- **Endpoint:** `/health`
+
+Responses with `200 OK` and a current status of peer as a JSON string:
+
+```json
+"Healthy"
+```
+
+### Pending Transactions
+
+TODO: if it is for _internal use only_, remove this endpoint from the docs?
+
+- **Protocol:** HTTP
+- **Encoding:** [Parity Scale Codec](#parity-scale-codec)
+- **Method:** `GET`
+- **Endpoint:** `/pending_transactions`
+
+_Internal use only._ Returns the transactions pending at the moment.
+
+## Endpoints for Status/Metrics
```

Comment:
do you think it makes sense to separate these from the other endpoints? the "api version" is neither a metric nor a status but it is a part of this section, while "health" is about a current status but is not included here

suggestion: what if we list all available endpoints alphabetically within the same "Endpoints" section? it'll be easier for users to find the info, and it'll create less confusion since no one is probably going to read this page from the start to the end, they'll look for info on a specific endpoint.

I see the benefits of having the status/metrics section, but we should make it less confusing, I think + it still makes sense to list endpoints alphabetically within each part

Reply:
The distinction between endpoints is made because they belong to different routers, e.g. 

```bash
# on 8080 port
"http://localhost:8080/transaction"

# on 9090 port
"http://localhost:9090/status"
```

API endpoints are configured with `torii.api_url` (which is required), status/metrics - with `torii.telemetry_url` (which is optional).

I don't see any reason to have separate routers for that, actually. I will ask the team soon to have a single one. However, since there is no decision yet, we need to document how Iroha works right now...

> what if we list all available endpoints alphabetically within the same "Endpoints" section?

Agree.

I think we need to clearly specify that some endpoint belongs to API router, and some to Telemetry router. And somewhere in the document describe the difference between them.

Reply:
Then we need "API Endpoints" and "Telemetry Endpoints" sections instead of the ones we currently have. And in the beginning of the page we can briefly explain what you can find in each section

Reply:
We discussed that we will merge the routers into a single one:

- https://github.com/hyperledger/iroha/issues/3950

Therefore, we can assume that there are no two separate routers.